src/scripts/utilities.mjs

```diff
@@ -36,3 +40,51 @@ export function removeMessageFromDomAndShiftOthers(messageToDelete) {
 	// Remove the deleted message
 	messageToDelete.remove();
 }
+
+/**
+ * Removes a specified message from the overlay, respecting themes' transitions.
+ *
+ * @param {string} messageId - unique identifier of message getting deleted
+ */
+export function removeMessage(messageId) {
+	const messageToDelete = document.querySelector(
+		`[data-twitch-message="${messageId}"]:not([data-twitch-message-display-status="deleting"])`
+	);
+
+	// Message has already been deleted through another means, or is in the process of getting deleted
+	if (!messageToDelete) return;
+
+	// Apply style hook for themes' outbound transitions
+	messageToDelete.setAttribute(
+		'data-twitch-message-display-status',
+		'deleting'
+	);
+
+	/**
+	 * Callback to remove message from DOM if it still exists.
+	 * Callback for addEventListener which cleans up after itself.
+	 */
+	function _callbackRemoveMessageFromDom() {
+		let remainingMessageToDelete = document.querySelector(
+			`[data-twitch-message="${messageId}"]`
+		);
+		if (remainingMessageToDelete) {
+			_removeMessageFromDomAndShiftOthers(messageToDelete);
+			removeEventListener('transitionend', _callbackRemoveMessageFromDom);
+			removeEventListener('animationend', _callbackRemoveMessageFromDom);
+		}
+	}
+
+	// Give animation a chance
+	messageToDelete.addEventListener(
+		'transitionend',
+		_callbackRemoveMessageFromDom
+	);
+	messageToDelete.addEventListener(
+		'animationend',
+		_callbackRemoveMessageFromDom
+	);
+
+	// If no animation, delete anyways
+	setTimeout(_callbackRemoveMessageFromDom, 1200);
```

Comment:
This feels a bit weird for themes without an animation.
I'd expect messages to disappear after the time I set in the urlbuilder on the homepage.
For themes with an animation: the animation starts at the time I set.
For themes without: they stay onscreen for the time I set + 1.2s

Reply:
I agree, this is an odd one for sure - but I'm not sure how to do it better. I suppose we could find a way to track "Was an animation or a transition started by applying the `[data-twitch-message-display-status="deleting"]` style hook?"

In practice, I think this weirdness will be mitigated by two things:

1. More themes having exit animations _(assuming those animations are less than 1.2 seconds long, which, tbf, they probably should be)_
2. Folks will likely choose longer message lifespans _(That extra 1.2 seconds matters less and less when your message lifespan is, say, 20 seconds than when it's 5 seconds)_.

That said, if we know of a way to do away with that 1.2 sec delay, I'm definitely open to it!

Reply:
@NickyMeuleman — I've been looking into this, and it looks like we could deduce whether there is an active transition or animation on the element.

Active transition: `!!messageToDelete.style.transition`
Active animation: Using the [`getAnimations`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAnimations) method which dropped in Chrome 84.

Using either of these seems like it could be fraught with heuristics, but it might be worth investigating.

Chrome 84 is more on the table now that folks who use OBS for Twitch are forced to update shortly, but I'd still like to hold off for a _bit_ before we start using `>75` APIs.

My proposal is that we ship this as it is, but file a new issue for revisiting this heuristic here a few months down the line. Thoughts?

Reply:
I agree, it's definitely a nice to have, but too soon at the moment. (TIL about that API, thanks for showing it to me, seems awesome!)
Keep as is, the confusion will be minimal (and only be there on themes that don't use animations, and probably only if the delay is short enough to notice).

(I didn't respond previously because I don't have a suggestion on how to handle this)

Reply:
I'm going to resolve this for now, and I'll create the issue once this PR is merged in!

Reply:
revisit that == '' check please. The question wheter that check serves the same purpose as the 
```
if (key == 'showCommands' && chosenValue === '') {
			// the showCommands option remained unchanged and should use the default
			chosenValue = defaultQueryParameters.showCommands;
}
```
Block.
Other than that minor detail, this looks great (I'd like to fold in that logic to the new check as well if that doesn't cause additional bugs I didn't think of). In other words: can we also delete that whole section now?